### PR TITLE
tests: proxy transport instead of disabling hostname verification

### PR DESCRIPTION
**Pull request by raboof**
Keep one test explicitly testing disabling hostname verification

### CODE REVIEW

File: akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala [hunk 1]
```diff
@@ -34,6 +32,7 @@ import scala.util.Success
 
 class GracefulTerminationSpec
   extends AkkaSpecWithMaterializer("""
+    akka.loglevel = debug
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala:35]: already set in `AkkaSpecWithMaterializer`

File: akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala [hunk 2]
```diff
@@ -199,7 +198,8 @@ class GracefulTerminationSpec
         super.basePoolSettings
           .withTransport(new ClientTransport {
             override def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[Http.OutgoingConnection]] = {
-              ClientTransport.TCP.connectTo(host, port, settings)
+              ExampleHttpContexts.proxyTransport(serverBinding.localAddress)
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala:201]: Does that change anything here? You could probably also just keep using `TCP.connectTo` and pass in the `serverBinding`'s host and port.

Comment (raboof) [reply akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala]: Good one, indeed

File: akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala (hunk 1 above)
Comment (raboof) [reply akka-http-core/src/test/scala/akka/http/scaladsl/GracefulTerminationSpec.scala]: oops

**Comment by raboof** [comment]
(travis is green but not reporting back)

**Pull request merged**